src/lib/operators/join_sort_merge.cpp

```diff
@@ -642,13 +669,15 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   * the table in reverse order. Returns the TablePosition of this element, and a satisfying element has been found.
   **/
   template <typename Function>
-  std::optional<TablePosition> _first_value_that_satisfies_reverse(
-      std::unique_ptr<MaterializedSegmentList<T>>& sorted_table, Function condition) {
-    for (size_t partition_id = sorted_table->size() - 1; partition_id < sorted_table->size(); --partition_id) {
-      auto partition = (*sorted_table)[partition_id];
-      if (!partition->empty() && condition((*partition)[0].value)) {
-        for (size_t index = partition->size() - 1; index < partition->size(); --index) {
-          if (condition((*partition)[index].value)) {
+  std::optional<TablePosition> _first_value_that_satisfies_reverse(const MaterializedSegmentList<T>& sorted_table,
+                                                                   const Function& condition) {
+    const auto sorted_table_size = sorted_table.size();
+    for (auto partition_id = sorted_table_size - 1; partition_id < sorted_table_size; --partition_id) {
```

Comment:
See above.

Reply:
Changed to
```
for (auto partition_iter = sorted_table.rbegin(); partition_iter != sorted_table.rend(); ++partition_iter) {
      if (!partition_iter->empty()) {
        return partition_iter->back().value;
      }
    }
```

Reply:
Changed to `for (auto partition_id = sorted_table_size - 1; partition_id >= 0; --partition_id) {`

Reply:
Nice ... I implemented `while (true)` ...

Reply:
Not really because `partition_id` can overflow, right? Did we use such loops before this PR or in any other places?

Reply:
Yeah, that's what I meant with the while loop. This was the reason of the segfault.

Using a cast to `int64_t` now. Not sure how to search for reverse loops in the code. :-/

Reply:
Searching for `; --` returns a few places but the only other loop where `<` is used as comparator is in the GroupKeyIndex.

Reply:
It's also using the "underflow". Don't like it. Definitely prefer a signed integer and a check for `i >= 0`.